src/components/Preferences/ModalActions.vue

```diff
@@ -4,6 +4,37 @@ import { mapState } from 'vuex';
 
 import Banner from '@/components/Banner.vue';
 
+interface Alert {
+  icon: string;
+  bannerText: string;
+  color: string;
+}
+
+interface AlertMap {
```

Comment:
So the keys can only ever be `reset`, `restart`, `error`, and empty string (in the case where none of the above applies)? I'm not sure we want the generic `[key: string]` case.

(Also, I think that means `AlertMap` simplifies to `Record<'reset' | 'restart' | 'error', Alert>`?)

Reply:
Thanks, I didn't quite like that implementation I went with. My change should be in the spirit of what you requested. Let me know what you think.